Sources/MapboxCoreNavigation/EventDetails.swift

```diff
@@ -95,7 +95,13 @@ struct NavigationEventDetails: EventDetails {
     let simulation: Bool
     let startTimestamp: Date?
     let sdkIdentifier: String
-    let sdkVersion: String = String(describing: Bundle.mapboxCoreNavigation.object(forInfoDictionaryKey: "CFBundleShortVersionString")!)
+    var sdkVersion: String {
+        if let stringForShortVersion = Bundle.mapboxCoreNavigation.object(forInfoDictionaryKey: "CFBundleShortVersionString") {
```

Comment:
I think we need to make `Bundle.mapboxCoreNavigation` work under SPM in general and not just for tests: #2922. Depending on the fix for that issue, the user default fallback may or may not be necessary.

Reply:
Yes. Because now the UserDefaults only added for test incase of the SPM generated Info.plist lacking certain keys.